src/webgldrawer.js

```diff
@@ -558,37 +558,35 @@
             let overlapFraction = this._calculateOverlapFraction(tile, tiledImage);
             let xOffset = tile.positionedBounds.width * overlapFraction.x;
             let yOffset = tile.positionedBounds.height * overlapFraction.y;
-
-            // x, y, w, h in viewport coords
             let x = tile.positionedBounds.x + (tile.x === 0 ? 0 : xOffset);
             let y = tile.positionedBounds.y + (tile.y === 0 ? 0 : yOffset);
             let right = tile.positionedBounds.x + tile.positionedBounds.width - (tile.isRightMost ? 0 : xOffset);
             let bottom = tile.positionedBounds.y + tile.positionedBounds.height - (tile.isBottomMost ? 0 : yOffset);
-            let w = right - x;
-            let h = bottom - y;
 
-            let matrix = new $.Mat3([
-                w, 0, 0,
-                0, h, 0,
-                x, y, 1,
+            const model = new $.Mat3([
+                right - x, 0, 0, // sx = width
+                0, bottom - y, 0, // sy = height
+                x, y, 1
             ]);
 
-            if(tile.flipped){
-                // flip the tile around the center of the unit quad
-                let t1 = $.Mat3.makeTranslation(0.5, 0);
-                let t2 = $.Mat3.makeTranslation(-0.5, 0);
-
-                // update the view matrix to account for this image's rotation
-                let localMatrix = t1.multiply($.Mat3.makeScaling(-1, 1)).multiply(t2);
-                matrix = matrix.multiply(localMatrix);
+            if (tile.flipped) {
+                // For documentation:
+                // // flip the tile around the center of the unit quad
+                // let t1 = $.Mat3.makeTranslation(0.5, 0);
+                // let t2 = $.Mat3.makeTranslation(-0.5, 0);
+                //
+                // // update the view matrix to account for this image's rotation
+                // let localMatrix = t1.multiply($.Mat3.makeScaling(-1, 1)).multiply(t2);
+                // matrix = matrix.multiply(localMatrix);
+
+                //Optimized: this works since matrix only contains main diagonal values & translation
+                model.scaleAndTranslateSelf(-1, 1, 1, 0);
             }
 
-            let overallMatrix = viewMatrix.multiply(matrix);
-
+            model.scaleAndTranslateOtherSetSelf(viewMatrix);
```

Comment:
Main optimization: this computes only 12 operations comapred to full matrix multiplication. It also replaces the model values internally (no new array creation). If you consider 50 tiles / frame, 30 frames / second, it saves at least 1500 array creations & tens of thousands of computations.